### PR TITLE
Port wiki pages to documentation website

### LINKED ISSUE #11299: Migrate pages in our GitHub wiki to the documentation website
Pages listed under https://github.com/matrix-org/synapse/wiki (especially the useful [synapse grafana dashboard guide](https://github.com/matrix-org/synapse/wiki/Understanding-Synapse-Performance-Issues-Through-Grafana-Graphs)) are currently quite undiscoverable. We should migrate them over to https://matrix-org.github.io/synapse.

https://github.com/matrix-org/synapse/tree/develop/docs#synapse-documentation has information on how to add pages to the documentation site.

### PR DESCRIPTION
**Pull request by H-Shay**
Fixes #11299. There was one file I did not port, it seemed out-of-date and perhaps not up to "official documentation" standards, but I am happy to change if others disagree.

### CODE REVIEW

File: docs/usage/administration/Admin_FAQ.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+```sql
+SELECT s.canonical_alias, g.room_id, count(*) AS num_rows 
+FROM 
+  state_groups_state AS g, 
+  room_stats_state AS s 
+WHERE g.room_id = s.room_id 
+GROUP BY s.canonical_alias, g.room_id
+ORDER BY num_rows desc 
+LIMIT 10;
+```
```
Comment (dklimpel) [inline docs/usage/administration/Admin_FAQ.md:96]:
~~~suggestion
```

You can also use the [List Room API](https://matrix-org.github.io/synapse/latest/admin_api/rooms.html#list-room-api)
and `order_by` `state_events`.
~~~

File: docs/usage/administration/Admin_FAQ.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+There are two exceptions when it might be sensible to delete your database and start again:
+* You have *never* joined any rooms which are federated with other servers. For instance, a local deployment which the outside world can't talk to. 
+* You are changing the `server_name` in the homeserver configuration. In effect this makes your server a completely new one from the point of view of the network, so in this case it makes sense to start with a clean database.
+(In both cases you probably also want to clear out the media_store.)
+
+I've stuffed up access to my room, how can I delete it to free up the alias?
+---
+Using the following curl command:
+
+    curl -H 'Authorization: Bearer <access-token>' -X DELETE https://matrix.org/_matrix/client/r0/directory/room/<room-alias>
```
Comment (dklimpel) [inline docs/usage/administration/Admin_FAQ.md:54]:
~~~suggestion
```
curl -H 'Authorization: Bearer <access-token>' -X DELETE https://matrix.org/_matrix/client/r0/directory/room/<room-alias>
```
~~~

File: docs/usage/administration/Admin_FAQ.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+\<access-token\> - can be obtained in riot by looking in the riot settings, down the bottom is:
```
Comment (dklimpel) [inline docs/usage/administration/Admin_FAQ.md:56]:
```suggestion
`<access-token>` - can be obtained in riot by looking in the riot settings, down the bottom is:
```

File: docs/usage/administration/Admin_FAQ.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+Access Token:\<click to reveal\> 
+
+\<room-alias\> - the room alias, eg. #my_room:matrix.org this possibly needs to be URL encoded also, for example  %23my_room%3Amatrix.org
```
Comment (dklimpel) [inline docs/usage/administration/Admin_FAQ.md:59]:
```suggestion
`<room-alias>` - the room alias, eg. #my_room:matrix.org this possibly needs to be URL encoded also, for example  %23my_room%3Amatrix.org
```

File: docs/usage/administration/Synapse-database-maintenance-tools.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+This blog post by Victor Berger explains how to use many of the tools listed on this page: https://levans.fr/shrink-synapse-database.html
+
+# List of useful tools and scripts for maintenance Synapse database:
+
+## [Purge Remote Media API](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/purge_remote_media.rst)
```
Comment (dklimpel) [inline docs/usage/administration/Synapse-database-maintenance-tools.md:5]:
```suggestion
## [Purge Remote Media API](https://matrix-org.github.io/synapse/latest/admin_api/media_admin_api.html#purge-remote-media-api)
```

File: docs/usage/administration/Synapse-database-maintenance-tools.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+The purge remote media API allows server admins to purge old cached remote media.
+
+#### Purge local media 
+
+There is no purge API for local media because you may be the only one with a copy. If you are sure you want to delete local media you could use something like the following to delete media that hasn't been accessed in a while.
```
Comment (dklimpel) [inline docs/usage/administration/Synapse-database-maintenance-tools.md:10]: This is not up to date.
There is an API: https://matrix-org.github.io/synapse/latest/admin_api/media_admin_api.html#delete-local-media

File: docs/usage/administration/Synapse-database-maintenance-tools.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+```
+find /path/to/synapse/media_store/local_content -atime +365 -delete
+find /path/to/synapse/media_store/local_thumbnails -atime +365 -delete
+```
+
+This will delete media that hasn't been accessed in 365 days.
+
+**Warning, check noatime flag**: You also have to double-check that the filesystem where synapse's media store don't have [noatime flag](https://unix.stackexchange.com/questions/219015/how-to-disable-access-time-settings-in-debian-linux/219017#219017). Check it with `mount`, `noatime` is something that is usually enabled by default to reduce read-write operations in the filesystem for a feature is not so demanded.
+
+## [Purge History API](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/purge_history_api.rst)
```
Comment (dklimpel) [inline docs/usage/administration/Synapse-database-maintenance-tools.md:21]:
```suggestion
## [Purge History API](https://matrix-org.github.io/synapse/latest/admin_api/purge_history_api.html)
```

File: docs/usage/administration/Understanding-Synapse-Performance-Issues-Through-Grafana-Graphs.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
+## Understanding Synapse through Grafana graphs
+
+It is possible monitor much of the internal state of Synapse using [Prometheus](https://prometheus.io) metrics and [Grafana](https://grafana.com/). A guide for configuring Synapse to provide metrics is available [here](https://github.com/matrix-org/synapse/blob/master/docs/metrics-howto.md) and information on setting up Grafana is [here](https://github.com/matrix-org/synapse/tree/master/contrib/grafana). In this setup, Prometheus will periodically scrape the information Synapse provides and store a record of it over time. Grafana is then used as an interface to query and present this information through a series of pretty graphs.
```
Comment (dklimpel) [inline docs/usage/administration/Understanding-Synapse-Performance-Issues-Through-Grafana-Graphs.md:3]:
```suggestion
It is possible monitor much of the internal state of Synapse using
[Prometheus](https://prometheus.io) metrics and [Grafana](https://grafana.com/).
A guide for configuring Synapse to provide metrics is available
[here](https://matrix-org.github.io/synapse/latest/metrics-howto.html)
and information on setting up Grafana is
[here](https://github.com/matrix-org/synapse/tree/master/contrib/grafana).
In this setup, Prometheus will periodically scrape the information Synapse provides and
store a record of it over time. Grafana is then used as an interface to query and
present this information through a series of pretty graphs.
```

File: docs/usage/administration/Understanding-Synapse-Performance-Issues-Through-Grafana-Graphs.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+## Garbage Collection
+
+![image](https://user-images.githubusercontent.com/1342360/82241911-da6ac180-9934-11ea-9a0d-a311fe22acd0.png)
+
+Large spikes in garbage collection times (bigger than shown here, I'm talking in the multiple seconds range), can cause lots of problems in Synapse performance. It's more an indicator of problems, and a symptom of other problems though, so check other graphs for what might be causing it.
+
+## Final Thoughts
+
+If you're still having performance problems with your Synapse instance and you've tried everything you can, it may just be a lack of system resources. Consider adding more CPU and RAM, and make use of [worker mode](https://github.com/matrix-org/synapse/blob/master/docs/workers.md) to make use of multiple CPU cores / multiple machines for your homeserver.
```
Comment (dklimpel) [inline docs/usage/administration/Understanding-Synapse-Performance-Issues-Through-Grafana-Graphs.md:72]:
```suggestion
If you're still having performance problems with your Synapse instance and you've tried
everything you can, it may just be a lack of system resources. Consider adding more CPU
and RAM, and make use of [worker mode](https://matrix-org.github.io/synapse/latest/workers.html)
to make use of multiple CPU cores / multiple machines for your homeserver.
```

**Comment by anoadragon453** [comment]
@dklimpel links leading to other parts of the documentation should be done through relative links to `.md` files. `mdbook` will then generate the right URLs when it builds the documentation.

Additionally, let's not try to review the content too much in this PR. Let's get it in in it's current state, and then iterate smaller fixes to the content from there.

**Comment by H-Shay** [comment]
@anoadragon453- is there anything you want me to pull from the review and add, or is this approved to merge in it's current state? Happy to make any changes necessary, just let me know what's best to proceed!

**Comment by dklimpel** [comment]
> Additionally, let's not try to review the content too much in this PR. Let's get it in in it's current state, and then iterate smaller fixes to the content from there.

I agree with you. But IMO if the documentation is moved, at least the links should be adjusted.

**Comment by squahtx** [comment]
A nit: could we lowercase the file names and convert them to `snake_case`?

**Comment by anoadragon453** [comment]
> @anoadragon453 -is there anything you want me to pull from the review and add, or is this approved to merge in it's current state? Happy to make any changes necessary, just let me know what's best to proceed!

Some things, yes. Sorry for the wait! I'm currently in the process of reviewing, and will request changes/approve then!

File: docs/other/Running-synapse-on-Single-board-computers.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+![Screenshot_2020-10-02_17-15-06](https://user-images.githubusercontent.com/71895/94945781-18771500-04d3-11eb-8419-83c2da73a341.png)
+
+This is #1211 and will also hopefully be mitigated by peeking matrix-org/matrix-doc#2753 so at least you don't need to wait for a join to complete before finding out if it's the kind of room you want. Note that you should first disable presence, otherwise it'll just make the situation worse #3120. There is a lot of database interaction too, so make sure you've [migrated your data](https://github.com/matrix-org/synapse/blob/master/docs/postgres.md) from the default sqlite to postgresql. Personally, I recommend patience - once the initial join is complete there's rarely any issues with actually interacting with the room, but if you like you can just block "large" rooms entirely.
```
Comment (anoadragon453) [inline docs/other/Running-synapse-on-Single-board-computers.md:23]:
```suggestion
This is #1211 and will also hopefully be mitigated by peeking matrix-org/matrix-doc#2753 so at least you don't need to wait for a join to complete before finding out if it's the kind of room you want. Note that you should first disable presence, otherwise it'll just make the situation worse #3120. There is a lot of database interaction too, so make sure you've [migrated your data](../postgres.md) from the default sqlite to postgresql. Personally, I recommend patience - once the initial join is complete there's rarely any issues with actually interacting with the room, but if you like you can just block "large" rooms entirely.
```

File: docs/usage/administration/Admin_FAQ.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+```
+grep 'GET-37' homeserver.log
+```
+
+If you want to paste that output into a github issue or matrix room, please remember to surround it with triple-backticks (```) to make it legible (see https://help.github.com/en/articles/basic-writing-and-formatting-syntax#quoting-code).
+
+
+What do all those fields in the 'Processed' line mean?
+---
+See https://matrix-org.github.io/synapse/latest/usage/administration/request_log.html.
```
Comment (anoadragon453) [inline docs/usage/administration/Admin_FAQ.md:81]:
```suggestion
See [Request log format](request_log.md).
```

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Thank you for doing this, and apologies again for the delay on review!

Some thoughts about visibility for contributors. Otherwise, I think the only changes to the content that are needed are ensuring all links to other bits of the documentation are relative links so that they render correctly on the site. That, and linking all "#xxxx" issues to the repo. I would leave behind any other fixes to the documentation, such as inaccuracies, and take those already suggested and include them in a subsequent PR. Just to limit the scope of what needs to be reviewed.

**Comment by H-Shay** [comment]
I've updated the relative links, PR/issue references, etc, renamed the relevant `.md` files using snake case and added the blurb to the changelog. I will create a follow-up PR to make the suggested changes from  @dklimpel (thank you for the review, btw!).

File: CHANGES.md [hunk 12]
```diff
@@ -18,7 +35,7 @@ The following issue is fixed in 1.47.1.
 
   Homeservers with the media repository disabled are unaffected. Homeservers configured with a federation whitelist are also unaffected.
 
-  Fixed by [91f2bd090](https://github.com/matrix-org/synapse/commit/91f2bd090).
+  Fixed by [91f2bd090](https://github.com/matrix-org/synapse/commit/91f2bd0
```
Comment (squahtx) [inline CHANGES.md:38]: Was this change intentional?

Comment (H-Shay) [reply CHANGES.md]: Not at all, thank you for catching that. I've fixed it.

File: CHANGES.md [hunk 13]
```diff
@@ -1,3 +1,19 @@
...
+onto the [documentation website](https://matrix-org.github.io/synapse/latest/).
+This was done for two reasons. The first was to ensure that changes are checked by 
+multiple authors before being committed (everyone makes mistakes!) and the second
+was visibility of the documentation. Not everyone knows that Synapse has some very
+useful information hidden away in its GitHub wiki pages. Bringing them to the 
+documentation website should help with visibility, as well as keep all Synapse documentation
+in one, easily-searchable location.
+
+Note that contributions to the documentation website happen through [GitHub pull 
+requests](https://github.com/matrix-org/synapse/pulls). Please visit #synapse-dev:matrix.org
```
Comment (anoadragon453) [inline CHANGES.md:14]:
```suggestion
requests](https://github.com/matrix-org/synapse/pulls). Please visit [#synapse-dev:matrix.org](https://matrix.to/#/#synapse-dev:matrix.org)
```

File: CHANGES.md [hunk 14]
```diff
@@ -1,3 +1,19 @@
+UNRELEASED
+==========
+
+We've decided to move the existing, somewhat stagnant pages from the GitHub wiki 
+onto the [documentation website](https://matrix-org.github.io/synapse/latest/).
```
Comment (anoadragon453) [inline CHANGES.md:5]:
```suggestion
to the [documentation website](https://matrix-org.github.io/synapse/latest/).
```

**Review by anoadragon453 — APPROVED** [review]
Some silly suggestions against my own words, but otherwise this lgtm.

Thanks for creating  #11466 as well!
